radio/src/lua/api_colorlcd.cpp

```diff
@@ -60,8 +60,11 @@ LcdFlags flagsRGB(LcdFlags flags)
   // RGB or indexed color?
   if (flags & RGB_FLAG)
     return flags;
-  else
-    return (flags & 0xFFFF) | COLOR(COLOR_VAL(flags)) | RGB_FLAG;
+
+  if (!(flags & 0xFF)) 
```

Comment:
I dunno if it's the right way to do it... but when I initially looked at this... `if (flags == 0)` seemed to work.

Reply:
Flags include font size information as well, so if(flags == 0) has no effect if font size is not zero.

Reply:
I thing that there is a much simpler solution:
https://github.com/EdgeTX/edgetx/blob/d775e75e81b35ea2729ef053ce34e205b6a967e4/radio/src/gui/colorlcd/colors.h#L29-L49
Delete `DEFAULT_COLOR_INDEX`, and now `COLOR_THEME_PRIMARY1_INDEX` is the new default!

Reply:
Of course, you need to grep for `DEFAULT_COLOR_INDEX` and replace a couple of appearances in the themes.

Reply:
I thought about it, but I'm not sure: I suspect the DEFAULT_COLOR_INDEX, was introduced for some corner cases, when the color table was not initialized yet (bootloader?). Since I do not know original considerations, I'm reluctant to do this "drastic" change. If anybody is sure this will not create problems - of course this is a cleaner solution.

Reply:
The bootloader does not use theme color (I know, because when I did, Raphael told me to not mess ever with that again 🤣)  I don't see any problems getting rid of `DEFAULT_COLOR_INDEX`. Here is the output from grep:
```
radio/src/gui/colorlcd/theme_manager.cpp:  if (i >= COLOR_COUNT) return DEFAULT_COLOR_INDEX;
radio/src/gui/colorlcd/theme_manager.cpp:  if (colorIndex >= DEFAULT_COLOR_INDEX && colorIndex < LCD_COLOR_COUNT) {
radio/src/gui/colorlcd/themes/480_default.cpp:      lcdColorTable[DEFAULT_COLOR_INDEX] = RGB(18, 94, 153);
radio/src/gui/colorlcd/themes/flysky.cpp:      lcdColorTable[DEFAULT_COLOR_INDEX] = RGB(18, 94, 153);
radio/src/gui/colorlcd/colors.h:  DEFAULT_COLOR_INDEX,
radio/src/gui/colorlcd/colors.h:#define COLOR(index) ((uint32_t)lcdColorTable[unsigned(index & 0xFF) >= LCD_COLOR_COUNT ? DEFAULT_COLOR_INDEX : unsigned(index & 0xFF)] << 16u)
```
As you can see, it is only used a few times (and they should probably be removed anyway).

Reply:
I will give it a try

Reply:
Actually, both variants do not work.
Now I understand the first comment: the font is always black now.